File: 백준/Silver/3758. KCPC/KCPC.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var in = bufio.NewScanner(os.Stdin)

func nextInt() int {
	in.Scan()
	n, _ := strconv.Atoi(in.Text())
	return n
}

func main() {
	in.Split(bufio.ScanWords)
	bw := bufio.NewWriter(os.Stdout)
	defer bw.Flush()

	T := nextInt()
	for T > 0 {
		T--
		n, k, ID, m := nextInt(), nextInt(), nextInt(), nextInt()
		table := make([][]int, n+1)
		scores, cnt, times := make([]int, n+1), make([]int, n+1), make([]int, n+1)
		for i := range table {
			table[i] = make([]int, k+1)
		}
		for i := 0; i < m; i++ {
			id, num, score := nextInt(), nextInt(), nextInt()
			times[id] = i
			cnt[id]++
			if table[id][num] < score {
				scores[id] -= table[id][num] - score
				table[id][num] = score
			}
		}
		rank := 1
		for i := 1; i <= n; i++ {
			if i == ID {
				continue
			}
			if scores[i] > scores[ID] {
				rank++
			} else if scores[i] == scores[ID] {
				if cnt[i] < cnt[ID] {
					rank++
				} else if cnt[i] == cnt[ID] && times[i] < times[ID] {
					rank++
				}
			}
		}
		fmt.Fprintln(bw, rank)
	}
}

File: 백준/Silver/3758. KCPC/KCPC.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder ans = new StringBuilder();
        int T = Integer.parseInt(br.readLine());
        for (int i = 0; i < T; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int n = Integer.parseInt(st.nextToken()), k = Integer.parseInt(st.nextToken())
                    , ID = Integer.parseInt(st.nextToken()), m = Integer.parseInt(st.nextToken());
            int[][] table = new int[n+1][k+1];
            int[] scores = new int[n+1], cnt = new int[n+1], times = new int[n+1];
            for (int j = 0; j < m; j++) {
                st = new StringTokenizer(br.readLine());
                int id  = Integer.parseInt(st.nextToken()), num = Integer.parseInt(st.nextToken())
                        , score = Integer.parseInt(st.nextToken());
                times[id] = j;
                cnt[id]++;
                if (table[id][num]<score) {
                    scores[id]-=table[id][num]-score;
                    table[id][num] = score;
                }
            }
            int rank = 1;
            for (int j = 1; j <= n; j++) {
                if (j==ID) continue;
                if (scores[j]>scores[ID]) rank++;
                else if (scores[j]==scores[ID]){
                    if (cnt[j]<cnt[ID]) rank++;
                    else if (cnt[j]==cnt[ID] && times[j]<times[ID]) rank++;
                }
            }
            ans.append(rank).append("\n");
        }
        System.out.println(ans);
    }
}